static/app/components/core/segmentedControl/index.tsx

```diff
@@ -66,17 +68,15 @@ export function SegmentedControl<Value extends string>({
 }: SegmentedControlProps<Value>) {
   const ref = useRef<HTMLDivElement>(null);
 
-  const collection = useCollection(props, collectionFactory);
-  const ariaProps: RadioGroupProps = {
+  const collection = useCollection(props as any, collectionFactory);
```

Comment:
this comes from extending `CollectionBase<any>`. According to the collection types, the `Value` type parameter needs to `extend object`, but ours extends string. No idea yet how this works, but at least I’ve plugged the `any` leak to consumers of this component and kept the type assertion internal.